urllib3/contrib/securetransport.py

```diff
@@ -207,9 +211,17 @@ def _read_callback(connection_id, data_buffer, data_length_pointer):
 
                 # We need to tell ctypes that we have a buffer that can be
                 # written to. Upsettingly, we do that like this:
-                chunk_size = base_socket.recv_into(
-                    buffer_view[read_count:requested_length]
-                )
+                if buffer_view is not None:
+                    chunk_size = base_socket.recv_into(
+                        buffer_view[read_count:requested_length]
+                    )
+                else:
+                    remaining = requested_length - read_count
+                    # This is likely innefficient, but works on Python 2.6
+                    buffer = (ctypes.c_char * remaining).from_address(
+                        data_buffer + read_count
+                    )
+                    chunk_size = base_socket.recv_into(buffer, remaining)
```

Comment:
If this works, surely we don't need the `buffer_view` at all?

Reply:
I don't know about the efficiency of constructing a new ctypes object from an address vs using a memoryview. I presume a memoryview is more efficient, just because less Python code is executing, and that ctypes isn't the most efficient way to do FFI.

Reply:
The reason I was using `memoryview` was to be able to use `recv_into`. In general, so long as `recv_into` is working (that is, we aren't writing data into an ephemeral buffer) it's probably worth avoiding the `memoryview` altogether for the sake of simplicity.

Reply:
Okay, I update it to just use the memory-offset ctypes array.